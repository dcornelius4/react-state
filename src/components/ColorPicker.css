.ColorPicker {
  button {
    border: none;
    padding: 2em;

    cursor: pointer;
  }

  .red {
    background-color: red;
  }

  .green {
    background-color: green;
  }

  .blue {
    background-color: blue;
  }
}