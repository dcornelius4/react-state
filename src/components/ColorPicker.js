import React, { PureComponent } from 'react';
import styles from './ColorPicker.css';

export default class ColorPicker extends PureComponent {
  state = {
    color: ''
  };

  colorHandler = color => {
    this.setState({ color });
  }

  render() {
    const { color } = this.state;

    return (
      <section className={styles.ColorPicker}>
        <button className={styles.red} onClick={this.colorHandler.bind(null, 'red')}>Red</button>
        <button className={styles.green} onClick={() => this.colorHandler('green')}>Green</button>
        <button className={styles.blue} onClick={() => this.colorHandler('blue')}>Blue</button>
        <div className={styles[color]}></div>
      </section>
    );
  }
}
