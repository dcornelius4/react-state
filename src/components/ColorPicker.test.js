import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker Component', () => {
  it('renders the color', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });
});
