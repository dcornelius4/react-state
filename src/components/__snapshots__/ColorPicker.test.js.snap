// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker Component renders the color 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="green"
    onClick={[Function]}
  >
    Green
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
</section>
`;
